### ISSUE TITLE
Missle Confusion

### ISSUE THREAD

**Issue by seedship** [ZeroK-RTS/Zero-K#4259]
So I was playing a game where my opponent dug a hole for his strategic target. I build a quake to counter his terraform, but it landed short of the target, in a way I couldn't explain. I looked at the replay to figure out what was happening.

I targeted the exact spot I wanted the quake to hit (red arc), but then I realized the trajectory computation made it land early, when the blue trajectory was what I wanted.
![image](https://user-images.githubusercontent.com/885667/103143003-08f01c80-470e-11eb-8cce-a4b5da3c1246.png)

I tested it out in single player, and it appears when trying to missle targets in holes, you have to aim behind the target because of the current missle trajectory. I think this is confusing, and if I known, I woulda been able to kill the target that game. I am thinking of 2 solutions.

1. Draw a line from the missile when you are targeting it, that will show the trajectory and estimated point of impact, given the player's knowledge of the map teraform state

2. Take the projection of the 2d location the player orders, and project it onto a flat horizontal plane, so the flight path is more intuitive (If I order the missle to go to point (X, Y, 0), it should pass through that point and land at like (X+e, Y+e, -20), not try to go to X,Y,-500 and crash early). This seems complicated and hard to generalize, but I'm just throwing ideas out there.

**Comment by seedship** [comment]
The replay if interested: http://zero-k.info/Battles/Detail/1016136 ([I]amNot[null]Bla1 (me) made a missle silo around 20 minutes in and tried missiling south's advanced geo)

**Comment by Arch-Shaman** [comment]
This could be a use case for my cruise missile gadget

**Comment by seedship** [comment]
That sounds nice, is it a 3rd party widget (like the lotus dancing thingy?) If so, where can I get it?

**Comment by GoogleFrog** [comment]
This sounds like something to add to the attack AoE widget for the missile silo missiles.

**Comment by Arch-Shaman** [comment]
No. It is not a widget. As I stated, it is a **gadget** -- meaning synced code. Widgets cannot alter projectile trajectories.
